fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -534,6 +547,36 @@ ACTOR Future<Void> commitBatch(
 	// Active load balancing runs at a very high priority (to obtain accurate estimate of memory used by commit batches) so we need to downgrade here
 	wait(delay(0, TaskPriority::ProxyCommit));
 
+	// Preliminary checks to address https://github.com/apple/foundationdb/issues/3647
+	for (int i = 0; i < trs.size();) {
+		bool hasMetadataMutations = false;
+		bool hasNormalMutations = false;
+		inspectTransaction(trs[i], &hasMetadataMutations, &hasNormalMutations);
+		if (hasMetadataMutations) {
+			if (hasNormalMutations) {
+				if (!trs[i].isLockAware()) {
+					// This transaction has metadata mutations, normal
+					// mutations, and is not lock aware. We can't make it
+					// lock-aware implicitly, since it has normal mutations. We
+					// can't let it commit either, since it has metadata
+					// mutations and isn't lock aware. Just reject it.
+					using std::swap;
+					swap(trs[i], trs.back());
```

Comment:
Calling `swap` unqualified is intentional. It's generally a good practice. If the type has a custom swap implementation then that will be selected instead.

In this case it doesn't really matter, since we know CommitTransactionRequest doesn't have a custom swap implementation, but it's a good habit.

Reply:
Nit: As this is a `cpp`-file I personally would prefer to add `using std::swap` to the top of the file (after the last include).

Reply:
Done